File: examples/python/basic.py
"""
Chart Generator API - Basic Usage Example

This example demonstrates the basic usage of the Chart Generator API.
API Documentation: https://docs.apiverve.com/ref/chartgenerator
"""

import os
import requests
import json

API_KEY = os.getenv('APIVERVE_API_KEY', 'YOUR_API_KEY_HERE')
API_URL = 'https://api.apiverve.com/v1/chartgenerator'

def call_chartgenerator_api():
    """
    Make a POST request to the Chart Generator API
    """
    try:
        # Request body
        request_body &#x3D; {
    &#x27;type&#x27;: &#x27;bar&#x27;,
    &#x27;data&#x27;: {
        &#x27;labels&#x27;: [
            &#x27;Q1&#x27;,
            &#x27;Q2&#x27;,
            &#x27;Q3&#x27;,
            &#x27;Q4&#x27;
        ],
        &#x27;datasets&#x27;: [
            {
                &#x27;label&#x27;: &#x27;Users&#x27;,
                &#x27;data&#x27;: [
                    50,
                    60,
                    70,
                    180
                ]
            },
            {
                &#x27;label&#x27;: &#x27;Revenue&#x27;,
                &#x27;data&#x27;: [
                    100,
                    200,
                    300,
                    400
                ]
            }
        ]
    }
}

        headers = {
            'x-api-key': API_KEY,
            'Content-Type': 'application/json'
        }

        response = requests.post(API_URL, headers=headers, json=request_body)

        # Raise exception for HTTP errors
        response.raise_for_status()

        data = response.json()

        # Check API response status
        if data.get('status') == 'ok':
            print('✓ Success!')
            print('Response data:', json.dumps(data['data'], indent=2))
            return data['data']
        else:
            print('✗ API Error:', data.get('error', 'Unknown error'))
            return None

    except requests.exceptions.RequestException as e:
        print(f'✗ Request failed: {e}')
        return None

if __name__ == '__main__':
    print('📤 Calling Chart Generator API...\n')

    result = call_chartgenerator_api()

    if result:
        print('\n📊 Final Result:')
        print(json.dumps(result, indent=2))
    else:
        print('\n✗ API call failed')

File: examples/javascript/basic.js
/**
 * Chart Generator API - Basic Usage Example
 *
 * This example demonstrates the basic usage of the Chart Generator API.
 * API Documentation: https://docs.apiverve.com/ref/chartgenerator
 */

const API_KEY = process.env.APIVERVE_API_KEY || 'YOUR_API_KEY_HERE';
const API_URL = 'https://api.apiverve.com/v1/chartgenerator';

/**
 * Make a POST request to the Chart Generator API
 */
async function callChartGeneratorAPI() {
  try {
    // Request body
    const requestBody &#x3D; {
    &quot;type&quot;: &quot;bar&quot;,
    &quot;data&quot;: {
        &quot;labels&quot;: [
            &quot;Q1&quot;,
            &quot;Q2&quot;,
            &quot;Q3&quot;,
            &quot;Q4&quot;
        ],
        &quot;datasets&quot;: [
            {
                &quot;label&quot;: &quot;Users&quot;,
                &quot;data&quot;: [
                    50,
                    60,
                    70,
                    180
                ]
            },
            {
                &quot;label&quot;: &quot;Revenue&quot;,
                &quot;data&quot;: [
                    100,
                    200,
                    300,
                    400
                ]
            }
        ]
    }
};

    const response = await fetch(API_URL, {
      method: 'POST',
      headers: {
        'x-api-key': API_KEY,
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(requestBody)
    });

    // Check if response is successful
    if (!response.ok) {
      throw new Error(`HTTP error! status: ${response.status}`);
    }

    const data = await response.json();

    // Check API response status
    if (data.status === 'ok') {
      console.log('✓ Success!');
      console.log('Response data:', data.data);
      return data.data;
    } else {
      console.error('✗ API Error:', data.error || 'Unknown error');
      return null;
    }

  } catch (error) {
    console.error('✗ Request failed:', error.message);
    return null;
  }
}

// Run the example
callChartGeneratorAPI()
  .then(result => {
    if (result) {
      console.log('\n📊 Final Result:');
      console.log(JSON.stringify(result, null, 2));
    }
  });
